src/client/common/experiments.ts

```diff
@@ -205,4 +227,25 @@ export class ExperimentsManager implements IExperimentsManager {
         }
         return true;
     }
+
+    /**
+     * Do best effort to download the experiments within timeout and use it in the current session only
+     */
+    public async doBestEffortToPopulateExperiments(): Promise<boolean> {
+        try {
+            const success = await Promise.race([
+                // Download and store experiments in the storage for the current session
+                this.downloadAndStoreExperiments(this.experimentStorage),
+                sleep(this.experimentEffortTimeout).then(() => {
```

Comment:
```suggestion
                sleep(this.experimentEffortTimeout).then(()=> false)
```

Please do not use `null` (against our coding standards).